### PR TITLE
Add optional min_version metadata to specify the minimum required cargo-c version

### PR DESCRIPTION
**Pull request by sdroege**
Fixes https://github.com/lu-zero/cargo-c/issues/99

----

This is on top of https://github.com/lu-zero/cargo-c/pull/103

### CODE REVIEW

**Comment by sdroege** [comment]
Merge conflicts resolved

**Review by lu-zero — APPROVED** [review]

**Comment by sdroege** [comment]
CI failure looks like some unrelated problem

**Comment by sdroege** [comment]
Also before merging this, I should probably add a section to the `README.md` about it. Let me just do that now.

File: README.md [hunk 1]
```diff
@@ -47,6 +47,13 @@ your crates, read [Building Crates so they Look Like C ABI Libraries][dev.to].
 ## Advanced
 You may override various aspects of `cargo-c` via settings in `Cargo.toml` under the `package.metadata.capi` key
 
+```toml
+[package.metadata.capi.header]
```
Comment (lu-zero) [inline README.md:51]:
```suggestion
[package.metadata.capi]
```

I think

Comment (sdroege) [reply README.md]: Indeed, thanks!

**Comment by sdroege** [comment]
CI failure is again some problem with lcov.

**Comment by lu-zero** [comment]
I disabled it for now, if you rebase it should stay green.

**Comment by sdroege** [comment]
Ok, rebased

**Comment by sdroege** [comment]
All green it seems, thanks :)

**Comment by lu-zero** [comment]
great :)

**Pull request merged**